### PR TITLE
Create temp. folders for certain tests

### PR DESCRIPTION
**Pull request by therearesomewhocallmetim**
Some tests fail when running in parallel because they compete for certain resources. To eliminate that situation we now create temporary folders for those tests and pass those folders as parameters.

### CODE REVIEW

**Comment by therearesomewhocallmetim** [comment]
@ygorshenin @gardster @syershov PTAL

File: tools/integration_tests_runner.py [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
+
+import logging
 import multiprocessing
+from optparse import OptionParser
+import tempfile
 from threading import Lock
 from threading import Thread
 import traceback
-import logging
+import shutil
```
Comment (gardster) [inline tools/integration_tests_runner.py]: sort please

File: tools/integration_tests_runner.py [hunk 2]
```diff
@@ -58,13 +64,29 @@ def exec_tests_in_queue(self):
                     return
 
                 test_file, test = self.tests.get()
-                self.exec_test(test_file, test)
+
+#                    test_file = '{} "--user-resource-path={}"'.format(test_file, tmpdir) #--data_path
```
Comment (gardster) [inline tools/integration_tests_runner.py]: What is this?

File: tools/integration_tests_runner.py [hunk 3]
```diff
@@ -58,13 +64,29 @@ def exec_tests_in_queue(self):
...
                 logging.error(traceback.format_exc())
 
 
-    def exec_test(self, test_file, test):
-        out, err, result = self.get_tests_from_exec_file(test_file, '--filter={test}'.format(test=test))
+    def exec_test(self, test_file, test, clean_env=False):
+        keys = '--filter={test}'.format(test=test)
+        if clean_env:
+            tmpdir = tempfile.mkdtemp()
+            logging.debug("Temp dir: {}".format(tmpdir))
```
Comment (gardster) [inline tools/integration_tests_runner.py]: Some of our systems has a old python interpreter. They can't handle {} string formatting. Please, use {0} instead.

**Comment by ygorshenin** [comment]
LGTM

**Comment by gardster** [comment]
LGTM

**Pull request merged**